unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/MouseCatcher/CursorLockHint.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace DCL
+{
+    public class CursorLockHint : MonoBehaviour
+    {
+        private bool hasBeenShown;
+
+        [SerializeField] private ShowHideAnimator toastRoot;
+        [SerializeField] private float duration;
+        
+        private Coroutine hideRoutine;
+
+        private void Start()
+        {
+            Utils.OnCursorLockChanged += HandleCursorLockChanges;
+        }
+
+        private void OnDestroy()
+        {
+            Utils.OnCursorLockChanged -= HandleCursorLockChanges;
+        }
+
+        private void HandleCursorLockChanges(bool isLocked)
+        {
+            if (!isLocked)
+            {
+                toastRoot.Hide();
+                return;
+            }
+            if (DataStore.i.camera.panning.Get()) return;
+            if (hasBeenShown) return;
+            if (toastRoot.isVisible) return;
+            toastRoot.gameObject.SetActive(true);
+            toastRoot.Show();
+            hasBeenShown = true;
+            HideToastAfterDelay();
+        }
+
+        private void HideToastAfterDelay()
+        {
+            if (hideRoutine != null)
+                StopCoroutine(hideRoutine);
+            hideRoutine = StartCoroutine(Wait(duration, () =>
+            {
+                toastRoot.Hide();
+                hideRoutine = null;
+            }));
+        }
+
+        private IEnumerator Wait(float delay, Action onFinishCallback)
```

Comment:
Maybe it could be fine to move this kind of utils functions to some common class in our project (for example the `Utils.cs` class) in order to have it available by anyone.

Reply:
done!